src/Authentication/RequestVerifier.php

```diff
@@ -28,7 +28,7 @@ public function authenticateRegistrationRequest(RequestInterface $request, strin
         );
     }
 
-    public function authenticatePostRequest(RequestInterface $request, ShopEntity $shop): void
+    public function authenticatePostRequest(RequestInterface $request, ShopInterface $shop): void
```

Comment:
In theory this would be a break, but in practice not so much because the shopEntity now implements the interface right?

Reply:
It is a break especially for other classes like the ShopRepositroy. The Bundle has no stable version yet so it's maybe okay. Should we better deprecate it for v1.0.0?

Reply:
I'm not sure how we will evolve this package in the future :thinking: What is the min php version?

i think we are php 8 right? then we could use a union type for now: https://php.watch/versions/8.0/union-types
```
public function authenticatePostRequest(RequestInterface $request, ShopEntity|ShopInterface $shop): void
```

and we can deprecate the usage of shopEntity for v1.0.0?

Reply:
It's only a problem for the interfaces because ShopEntity implements ShopInterface as you already said. But with PHP 8.0 it seems like it is no problem if you implement an interface by accepting only one type of an union type.

```
interface WithUnionType {
   public function a(Foo|Bar $a);
}

class CorrectImplementation implementes WithUnionType {
   public function a(Foo $a);   
}
```

Reply:
As the AppTemplate is currently requiring dev-main there aren't any promises of avoiding BC breaking changes, so you should not worry about that. If you want to make sure, start introducing versioning (0.x is fine too) and adjust the AppTemplate accordingly.

Reply:
I think the Union solution allows the most flexibility without adding too much overhead. I agree we should start proper versioning soon.